### DISCUSSION TITLE
Automatic insert/update of TXT file property

### DISCUSSION THREAD

**Discussion by ZimTom** [zim-desktop-wiki/zim-desktop-wiki#2101] [Ideas]
Hello all,
in case this "idea" already has a beard, I have to apologize, but I am new here and after looking around I could not find anything matching - which doesn't mean that it might not be somewhere.

I am using ZIM to rebuild and extend my personal information archive and as a habit I always write manually the date of my latest modification in each document at the very first line under the headline.
I have tried the option Insert | Data & Time but the only thing I can do with it is either insert a date or complete timestamp as static text or creating a journal entry by using the option Link Date.

I wonder if there is a function that allows putting in a kind of link in the wiki file which just displays the Last Modified Date (or complete timestamp) of the txt file itself when accessing it and - if not - if this would be possible and worthwhile to be integrated in ZIM.

**Comment by ZimTom** [comment]
I am sorry, but there was a mishap with the thread title.
It was intended to be ..."TXT file property"
If somebody with an adequate user role admin is able and willing to correct that, this would be great.
Apologies to anyone considering the crippled title as misleading and/or confusing :-)

**Comment by jaap-karssenberg** [comment]
Do you mean to just display it or make it part of the file content ?

My own way of working is to use the "insert date" to make timestamp part of the file content. That way I keep a log of updates per topic. But you already describe that is not what you are looking for, so I assume you mean to display the file system modified property without making it part of the text - correct ?

Currently this is not implemented, but straight-forward from a plugin point of view. It could either be a label in the UI (outside the text area) - e.g. in the statusbar or title bar of the window - or implemented as an inline object (like "fields" in e.g. MS Word). Either way the plugin gets the modified date of the page source file and inserts it in a UI widget.

--- Jaap

**Reply by ZimTom** [reply]
Hi Jaap,
thanks for your comment as well as the corrrection of the thread title. At the moment I do the same as you do. I simply write the date under the first headline (document title) as part of the content. But I thought it might be a good idea to be able to see information like file creation date, last modified date etc. when viewing the Wiki. From my understanding something like the fields in MS Word would be great, as it would fulfil the main aspects I thought about:

1. The information is visbible when viewing a file via ZIM directly
2. The Information is updated automatically
3. Assumming that it would work similarly to the filelds in MS Word the information would integrated in the document content as a kind of snapshot, so that it would also be visible in the HTML view as well as in exported files or file structures.

Of course the manual way always works. I just thought this might be something others would probably fancy as well :-)
